clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -495,7 +496,14 @@ public ApiException build(String message) {
             public ApiException build(String message) {
                 return new OperationNotAttemptedException(message);
             }
-        });
+    }),
+    KAFKA_STORAGE_ERROR(56, "Disk error when trying to access log file on the disk.",
```

Comment:
Now I understand the question.

I am not sure we should make `UnknownServerException` a retriable error because I am a bit concerned with having unnecessary retry if we do that. I think we can keep the current practice by making an error retriable only if we know it should be retriable. Is there any reason that we should make `UnknownServerException` retriable?

I understand that `KafkaStorageException` was previously non-retriable because it is send to client as `UnknownServerException`. But I couldn't find anything wrong with making it retriable now. It also doesn't break any contract between client and server because client is now receiving a new error (i.e. KafkaStorageError) from client's perspective. Is there any concern with this approach?